cola/widgets/remote.py

```diff
@@ -116,6 +123,42 @@ def task(self):
         return self.model_action(self.remote, **self.kwargs)
 
 
+def _emit_push_notification(selected_remotes, pushed_remotes):
+    if notifypy is None:
+        return
+
+    notification = notifypy.Notify()
+
+    total_count = len(selected_remotes)
+    success_count = len(pushed_remotes)
+
+    if success_count == total_count:
+        notification.title = N_('Push Successful')
+        notification.icon = resources.package_data('icons', 'git-cola.svg')
+
+        notification.message = N_(
+            f'All commits pushed to remotes: {", ".join(pushed_remotes)}'
+        )
+    elif success_count != 0:
+        notification.title = N_('Push Partially Successful')
+        notification.icon = resources.package_data('icons', 'git-cola-error.svg')
+
+        notification.message = N_(
+            f'Commits pushed to {success_count} remotes {", ".join(pushed_remotes)}\n'
+            f'Failed to push commits to {total_count - success_count} remotes '
+            f'{", ".join(list(set(selected_remotes) - set(pushed_remotes)))}'
+        )
+    else:
+        notification.title = N_('Push Failed')
+        notification.icon = resources.package_data('icons', 'git-cola-error.svg')
+
+        notification.message = N_(
+            f'No commits pushed to remotes: {", ".join(selected_remotes)}'
+        )
```

Comment:
Thanks! This is looking close enough. I'll probably merge this after making a small fixup commit locally.

The internationalization methods translate fixed strings so we can't pass anything varying to them. The way it's used is to include python string `%(variable)` references in the strings and those references are actually part of the translated string. We then use the old-school python `%` format operator to format the output from `N_(...)` to fill in the variables.

I'll be adjusting the titles and messages. I'm going to make the titles match what I suggested because it's more information dense to include the remote counts in the title and to use a consistent (unchanging) title for all of the scenarios above.

Also, I'm going to pass the pushed a non-pushed remotes as separate variables so that we don't have to do fancy `set(...)` operations. It's better for the code to be simple and dumb.

Lastly, I'm going to adjust the messages to be more succinct with just `Pushed:` and `Not pushed:` notes. The reason is that we want to be as brief as possible (they're notifications) and also because, at least on this desktop environment, the message can only be a single line so `\n` newline separators don't visualize as expected. That's why I had mentioned `\t` originally -- using a tab did provide a visual separator in my testing and was recognized by the notification display.

The other reason why having a consistent title (1) and just the "Pushed" (2) and "Not pushed" (3) notes is that it minimizes the number of strings that need translation. It would only use 3 different strings. The current code above uses 6 different distinct strings that would need to be translated separately.

Hope that makes sense. Thanks again for getting things this far.

I'll be merging shortly after applying this fixup locally.